packages/babel-plugin-proposal-optional-chaining/test/util.test.js

```diff
@@ -0,0 +1,85 @@
+import { willPathCastToBoolean } from "../src/util";
+import { parseSync, traverse } from "@babel/core";
+
+function getPath(input, parserOpts) {
+  let targetPath;
+  traverse(parseSync(input, { parserOpts, filename: "example.js" }), {
+    OptionalMemberExpression(path) {
+      targetPath = path;
+      path.stop();
+    },
+    noScope: true,
+  });
+  return targetPath;
+}
+
+describe("willPathCastToBoolean", () => {
+  const positiveCases = [
+    "if(a?.b) {}",
+    "while(a?.b) {}",
+    "a?.b ? 0 : 1",
+    "for(;a?.b;);",
+    "while(a?.b);",
+    "do {} while (a?.b)",
+    "!a?.b",
+    "if(a && a?.b) {}",
+    "if(a?.b && a) {}",
+    "while(a || a?.b) {}",
+    "while(a?.b || a) {}",
+    "for(; a ?? a?.b ;);",
+    "do {} while (0, a?.b)",
+    "(a?.b, 0)",
+    "!(a && ( b || ( c ?? ( a?.b && d ) ) ) )",
+
+    // parenthesized
+    "!((a?.b), 0)",
+    "((a?.b)) ? 0 : 1",
+    "while( (a || ((a?.b) && c) ) );",
+  ];
+
+  const negativeCases = [
+    "() => a?.b",
+    "for(; a?.b ?? a;);",
+    "a?.b && a",
+    "a && a?.b",
+    "() => a?.b || a",
+    "() => a || a?.b",
+    "a?.b()",
+    "a = a?.b && null",
+
+    // parenthesized
+    "(a?.b)",
+  ];
+
+  describe("default parser options", () => {
+    test.each(positiveCases.map(x => [x]))(
```

Comment:
What is the `.map(x => [x])` needed for?

Reply:
Ahh Jest will internally map 1-D array to 2-D: https://jestjs.io/docs/en/api#1-testeachtablename-fn-timeout

will update later.